vnda/loaders/productListingPage.ts

```diff
@@ -63,34 +64,67 @@ const searchLoader = async (
   const term = props.term || props.slug || qQueryString ||
     undefined;
 
+  const categoryTagName = props.term || url.pathname.split("/").slice(1) || "";
+
+  const categoryTagNames = Object.values(
+    Object.fromEntries(new URL(req.url).searchParams.entries()),
+  );
+
+  const promises = categoryTagNames.concat(
+    [...categoryTagName].filter((isUndefined) =>
+      isUndefined !== undefined
+    ) as string[],
+  ).map((categoryTagName) => {
+    return api["GET /api/v2/tags/:name"]({ name: categoryTagName }, STALE)
+      .then((res) => res.json())
+      .catch(() => undefined);
+  });
+
+  const tags = await Promise.all(promises);
+
+  const seoCategories = tags.slice(-[...categoryTagName].length).filter((
+    tag,
+  ): tag is { name: string; title: string; subtitle: string } =>
+    tag !== undefined
+  );
+
+  const categories = seoCategories.map(({ name }) => name);
+
+  const resolvedTagNames = tags
+    .filter((tag): tag is { name: string } => tag !== undefined)
```

Comment:
prepare if tag has no name.

```suggestion
    .filter((tag): tag is { name?: string } => tag !== undefined && tag.name)
```